pkg/controller/nodemanager/starter.go

```diff
@@ -0,0 +1,146 @@
+	"k8s.io/cloud-provider-gcp/pkg/controller/nodeipam/ipam"
+	"k8s.io/cloud-provider/controllers/node"
+	"k8s.io/cloud-provider/controllers/nodelifecycle"
+	"k8s.io/klog/v2"
+)
+
+// startScopedNodeControllers runs a new OSS Node Controller in a goroutine.
+// This goroutine blocks until the provided context is canceled.
+func (m *NodeManagerController) startScopedNodeControllers(ctx context.Context, cancel context.CancelFunc, pc *v1.ProviderConfig, pcKey string) {
+	klog.Infof("[%s] Attempting to start scoped node controller", pcKey)
+	var filteredFactory *FilteredSharedInformerFactory
+
+	// Ensure cleanup happens when this function exits (either due to error or context cancellation)
+	defer func() {
+		klog.Infof("[%s] Scoped node controller stopping", pcKey)
+		// Cancel the context to stop other controllers (IPAM, Lifecycle)
+		cancel()
+
+		// Cleanup event handlers from the global informer
+		if filteredFactory != nil {
+			klog.Infof("[%s] Cleaning up filtered informer event handlers", pcKey)
+			filteredFactory.Cleanup()
+		}
+
+		// Ensure the runningControllers map is cleaned up
+		m.runningControllersLock.Lock()
+		if _, exists := m.runningControllers[pcKey]; exists {
+			klog.V(2).Infof("[%s] Cleaning up controller entry from running map", pcKey)
+			delete(m.runningControllers, pcKey)
+		}
+		m.runningControllersLock.Unlock()
+	}()
+
+	// Create the new, scoped GCECloud object
+	klog.V(2).Infof("[%s] Creating tenant-scoped GCE cloud object...", pcKey)
+	scopedCloud, err := CreateTenantScopedGCECloud(m.config, pc)
+	if err != nil {
+		klog.Errorf("[%s] Failed to create scoped cloud: %v. Aborting controller startup.", pcKey, err)
+		return
+	}
+	klog.V(2).Infof("[%s] Scoped GCE cloud created successfully", pcKey)
+
+	providerConfigName, err := getNodeLabelSelector(pc)
+	if err != nil {
+		klog.Errorf("[%s] Failed to get node label selector: %v. Aborting controller startup.", pcKey, err)
+		return
+	}
+	klog.Infof("[%s] Using node label selector: %s", pcKey, providerConfigName)
+
+	klog.V(2).Infof("[%s] Creating filtered informer factory...", pcKey)
+	filteredFactory = NewFilteredSharedInformerFactory(m.mainInformerFactory, ProviderConfigLabelKey, providerConfigName)
+
+	klog.Infof("[%s] Waiting for main informer caches to sync...", pcKey)
+	if !cache.WaitForCacheSync(ctx.Done(),
+		m.mainInformerFactory.Core().V1().Nodes().Informer().HasSynced,
+		m.mainInformerFactory.Core().V1().Pods().Informer().HasSynced,
+	) {
+		klog.Errorf("[%s] Failed to sync main caches. Aborting controller startup.", pcKey)
+		return
+	}
+	klog.Infof("[%s] Main informer caches synced successfully", pcKey)
+
+	klog.Infof("[%s] Creating OSS Cloud Node Controller...", pcKey)
+	nodeController, err := node.NewCloudNodeController(
+		filteredFactory.Core().V1().Nodes(),
+		m.kubeClient,
+		scopedCloud,
+		m.config.ComponentConfig.NodeStatusUpdateFrequency.Duration,
+		m.config.ComponentConfig.NodeController.ConcurrentNodeSyncs,
+	)
+	if err != nil {
```

Comment:
there has to be two independent launchers, and users should opt-in via the standards flags to enable one or other controllers, check this for prior art https://github.com/kubernetes/cloud-provider-gcp/pull/895/files

The litmus test is that the addition of this new controller/folder will have zero impact, and the behavior is governed by the flags